functions.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+function displayCollection(array $allCards) : string {
+    $collection = '';
+    foreach ($allCards as $card) {
+        $collection .=
+            '<span>________________________________________</span>' .
+            '<h2>' . $card['name'] . '</h2>' .
+            '<img src="' . $card['image_link'] . '" alt="an image of the card">' .
+            '<p>Unit Strength: ' . $card['unit_strength'] . '</p>' .
+            '<p>Type: ' . $card['type'] . '</p>' .
+            '<p>Price: ' . $card['price'] . '</p>' .
+            '<p>Territory: ' . $card['territory'] . '</p>';
+    }
+    if($allCards === [['name' => 3.14, 'image_link' => 3.14, 'unit_strength' => 'PIE', 'type' => 3.14, 'price' => 'PIE', 'territory' => 3.14]]) {
+        return 'PIE IN YOUR FACE';
```

Comment:
this is saying that if the function receives this _exact_ incorrect input, then it returns an error message. it doesn't take care of _all_ possible incorrect inputs, it just chooses the specific one that happens to be in the unit test. 

in the future, someone might edit your code and, for example, accidentally pass the wrong array into the function. your function would end up throwing an error that would be displayed on the page plainly for users to see.

you need to handle all possible invalid arrays (valid arrays should have all the keys you need to create the html).

Reply:
also unfortunately you need to put a more informative (boring) error message

Reply:
fixed